src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -1009,22 +963,109 @@ def ramp_fit_slopes(ramp_data, gain_2d, readnoise_2d, save_opt, weighting):
     if pixeldq_sect is not None:
         del pixeldq_sect
 
-    # Compute the final 2D array of differences; create rate array
-    median_diffs_2d /= n_int
-    med_rates = median_diffs_2d / group_time
-
-    del median_diffs_2d
-    del first_diffs_sect
 
     ramp_data.data = data
     ramp_data.err = err
     ramp_data.groupdq = groupdq
     ramp_data.pixeldq = inpixeldq
 
+
     return max_seg, gdq_cube_shape, effintim, f_max_seg, dq_int, num_seg_per_int,\
         sat_0th_group_int, opt_res, pixeldq, inv_var, med_rates
 
 
+def compute_median_rates(ramp_data):
+    """
+    Compute the median rates for each pixel.
+
+    Parameter
+    ---------
+    ramp_data : RampData
+        Contains the data needed to compute the median rates.
+
+    median_rates : ndarray
+        The computed median rates for each pixel.
+    """
+    nints, ngroups, nrows, ncols = ramp_data.data.shape
+    imshape = (nrows, ncols)
+
+    group_time = ramp_data.group_time
+    frame_time = ramp_data.group_time
```

Comment:
Should the RHS be ramp_data.frame_time ?

Reply:
Yes.  I fixed that.